account/views.py

```diff
@@ -46,31 +53,25 @@ def login():
     return render_template('account/login.html', form=form)
 
 
+
+def create_user(form):
+    user = User(
+        first_name=form.first_name.data,
+        last_name=form.last_name.data,
+        email=form.email.data,
+        password=form.password.data)
+    db.session.add(user)
+    db.session.commit()
+    return user
+
 @account.route('/register', methods=['GET', 'POST'])
 def register():
     """Register a new user, and send them a confirmation email."""
     form = RegistrationForm()
     if form.validate_on_submit():
-        user = User(
-            first_name=form.first_name.data,
-            last_name=form.last_name.data,
-            email=form.email.data,
-            password=form.password.data)
-        db.session.add(user)
-        db.session.commit()
+        user = create_user(form)
         token = user.generate_confirmation_token()
-        confirm_link = url_for('account.confirm', token=token, _external=True)
-        get_queue().enqueue(
-            send_email,
-            recipient=user.email,
-            subject='Confirm Your Account',
-            template='account/email/confirm',
-            user=user,
-            confirm_link=confirm_link)
-        flash('A confirmation link has been sent to {}.'.format(user.email),
-              'warning')
-        return redirect(url_for('main.index'))
-    return render_template('account/register.html', form=form)
```

Comment:
The register() function could be improved by separating  the user creation logic from the  routing logic. This would make the code more modular and easier to test